lib/frame.js

```diff
@@ -419,105 +532,106 @@ function _filterSubject(state, subject, frame, flags) {
-    }
-
-    // no longer a wildcard pattern if frame has any non-keyword properties
-    wildcard = false;
-
-    // skip, but allow match if node has no value for property, and frame has a
-    // default value
-    if(nodeValues.length === 0 && hasDefault) {
+      if(!flags.requireAll) {
+        return matchThis;
+      }
+    } else if(isKeyword(key)) {
       continue;
-    }
+    } else {
+      // Force a copy of this frame entry so it can be manipulated
+      const thisFrame = util.getValues(frame, key)[0];
+      let hasDefault = false;
+      if(thisFrame) {
+        _validateFrame([thisFrame]);
+        hasDefault = '@default' in thisFrame;
+      }
 
-    // if frame value is empty, don't match if subject has any value
-    if(nodeValues.length > 0 && isEmpty) {
-      return false;
-    }
+      // no longer a wildcard pattern if frame has any non-keyword properties
+      wildcard = false;
 
-    if(thisFrame === undefined) {
-      // node does not match if values is not empty and the value of property
-      // in frame is match none.
-      if(nodeValues.length > 0) {
+      // skip, but allow match if node has no value for property, and frame has a
+      // default value
+      if(nodeValues.length === 0 && hasDefault) {
+        continue;
+      }
+
+      // if frame value is empty, don't match if subject has any value
+      if(nodeValues.length > 0 && isEmpty) {
         return false;
       }
-      matchThis = true;
-    } else if(types.isObject(thisFrame)) {
-      // node matches if values is not empty and the value of property in frame
-      // is wildcard
-      matchThis = nodeValues.length > 0;
-    } else {
-      if(graphTypes.isValue(thisFrame)) {
-        // match on any matching value
-        matchThis = nodeValues.some(nv => _valueMatch(thisFrame, nv));
-      } else if(graphTypes.isSubject(thisFrame) ||
-        graphTypes.isSubjectReference(thisFrame)) {
-        matchThis =
-          nodeValues.some(nv => _nodeMatch(state, thisFrame, nv, flags));
-      } else if(graphTypes.isList(thisFrame)) {
-        const listValue = thisFrame['@list'][0];
-        if(graphTypes.isList(nodeValues[0])) {
-          const nodeListValues = nodeValues[0]['@list'];
-
-          if(graphTypes.isValue(listValue)) {
-            // match on any matching value
-            matchThis = nodeListValues.some(lv => _valueMatch(listValue, lv));
-          } else if(graphTypes.isSubject(listValue) ||
-            graphTypes.isSubjectReference(listValue)) {
-            matchThis = nodeListValues.some(lv => _nodeMatch(
-              state, listValue, lv, flags));
+
+      if(thisFrame === undefined) {
+        // node does not match if values is not empty and the value of property
+        // in frame is match none.
+        if(nodeValues.length > 0) {
+          return false;
+        }
+        matchThis = true;
+      } else if(types.isObject(thisFrame)) { // XXX only framing keywords
```

Comment:
Does this "XXX" comment need to be addressed?

Reply:
It was in my code, and it could relate to some of the yet-to-be fixed issues.